packages/frontend/lib/structured-data.ts

```diff
@@ -0,0 +1,49 @@
+interface PublisherType {
+    '@type': string;
+    name: string;
+    url: string;
+    logo: string;
+    sameAs: string[];
+}
+
+interface PotentialActionType {
+    '@type': string;
+    target: string;
+}
+
+export interface StructuredDataType {
+    '@id': string;
+    '@type': string;
+    '@context': string;
+    isAccessibleForFree: boolean;
+    publisher: PublisherType;
+    potentialAction: PotentialActionType;
+}
+
+export function getStructuredData(
+    guardianBaseURL: string,
+    webURL: string,
+): StructuredDataType {
+    return {
+        '@id': webURL,
+        '@type': 'NewsArticle',
+        '@context': 'http://schema.org',
+        publisher: {
+            '@type': 'Organization',
+            name: 'The Guardian',
+            url: guardianBaseURL,
+            logo:
+                'https://assets.guim.co.uk/images/favicons/fee5e2d638d1c35f6d501fa397e53329/152x152.png',
+            sameAs: [
+                'https://www.facebook.com/theguardian',
+                'https://twitter.com/guardian',
+                'https://www.youtube.com/user/TheGuardian',
+            ],
+        },
+        isAccessibleForFree: true,
+        potentialAction: {
+            '@type': 'ViewAction',
+            target: `android-app://com.guardian/${webURL.replace('://', '/')}`,
+        },
+    };
+}
```

Comment:
I wonder if we should pass [this data from frontend](https://github.com/guardian/frontend/blob/master/common/app/model/meta/LinkedData.scala) instead of duplicating it here? We should at least keep a note of the duplication in the [duplications doc](https://github.com/guardian/dotcom-rendering/blob/master/docs/duplication.md).

Reply:
That's a good point.

I was imagining that the equivalent (duplicate) `frontend` data would later be removed along with  the Scala templates.

I wonder where the right place for this is structured data. In my mind it's in this rendering project, being representation markup. But I'm all for avoiding duplication!

Reply:
Okay, on balance I'm inclined towards avoiding duplication and having frontend be the source of truth for this data. It'll be a long time before we remove those Scala templates. That logo URL looks a bit brittle also!

However I don't want this to block you, so I'll add this task to our backlog.

Reply:
@SiAdcock I've updated this PR to just render `structuredData` which is passed down to it from `frontend`.

The `frontend` PR is here https://github.com/guardian/frontend/pull/20951